extensions/common/vault/hashicorp/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/client/HashicorpVaultTokenRenewService.java

```diff
@@ -174,48 +156,31 @@ public Result<Long> renewToken() {
             return Result.failure("Failed to renew token with reason: %s".formatted(e.getMessage()));
         }
     }
-
-    @NotNull
-    private HttpUrl getHealthCheckUrl() {
-        var vaultHealthPath = settings.healthCheckPath();
-        var isVaultHealthStandbyOk = settings.healthStandbyOk();
-
-        // by setting 'standbyok' and/or 'perfstandbyok' the vault will return an active
-        // status
-        // code instead of the standby status codes
-
-        return HttpUrl.parse(settings.url())
-                .newBuilder()
-                .addPathSegments(PathUtil.trimLeadingOrEndingSlash(vaultHealthPath))
-                .addQueryParameter("standbyok", isVaultHealthStandbyOk ? "true" : "false")
-                .addQueryParameter("perfstandbyok", isVaultHealthStandbyOk ? "true" : "false")
-                .build();
-    }
-
+    
     @NotNull
     private Request httpGet(HttpUrl requestUri) {
         return new Request.Builder()
                 .url(requestUri)
-                .headers(headers)
+                .headers(getHeaders(tokenProvider.vaultToken()))
                 .get()
                 .build();
     }
-
+    
     @NotNull
     private Request httpPost(HttpUrl requestUri, Object requestBody) {
         return new Request.Builder()
                 .url(requestUri)
-                .headers(headers)
+                .headers(getHeaders(tokenProvider.vaultToken()))
                 .post(createRequestBody(requestBody))
                 .build();
     }
-
-    private Headers getHeaders() {
+    
+    private Headers getHeaders(String token) {
```

Comment:
given that `tokenProvider` is a class field, maybe it could be used internally in this method instead of having the token passed every time